Kernel/Graphics/GraphicsManagement.cpp

```diff
@@ -45,45 +44,43 @@ UNMAP_AFTER_INIT GraphicsManagement::GraphicsManagement()
 
 void GraphicsManagement::disable_vga_emulation_access_permanently()
 {
-    SpinlockLocker locker(m_main_vga_lock);
-    disable_vga_text_mode_console_cursor();
-    IO::out8(0x3c4, 1);
-    u8 sr1 = IO::in8(0x3c5);
-    IO::out8(0x3c5, sr1 | 1 << 5);
-    microseconds_delay(1000);
-    m_vga_access_is_disabled = true;
+#if ARCH(I386) || ARCH(X86_64)
+    if (!m_vga_arbiter)
+        return;
+    m_vga_arbiter->disable_vga_emulation_access_permanently({});
+#endif
 }
 
 void GraphicsManagement::enable_vga_text_mode_console_cursor()
 {
-    SpinlockLocker locker(m_main_vga_lock);
-    if (m_vga_access_is_disabled)
+#if ARCH(I386) || ARCH(X86_64)
+    if (!m_vga_arbiter)
         return;
-    IO::out8(0x3D4, 0xA);
-    IO::out8(0x3D5, 0);
+    m_vga_arbiter->enable_vga_text_mode_console_cursor({});
+#endif
 }
 
 void GraphicsManagement::disable_vga_text_mode_console_cursor()
 {
-    SpinlockLocker locker(m_main_vga_lock);
-    if (m_vga_access_is_disabled)
+#if ARCH(I386) || ARCH(X86_64)
+    if (!m_vga_arbiter)
         return;
-    IO::out8(0x3D4, 0xA);
-    IO::out8(0x3D5, 0x20);
+    m_vga_arbiter->disable_vga_text_mode_console_cursor({});
+#endif
 }
 
+#if ARCH(I386) || ARCH(X86_64)
 void GraphicsManagement::set_vga_text_mode_cursor(size_t console_width, size_t x, size_t y)
 {
-    SpinlockLocker locker(m_main_vga_lock);
-    if (m_vga_access_is_disabled)
+    if (!m_vga_arbiter)
         return;
-    enable_vga_text_mode_console_cursor();
-    u16 value = y * console_width + x;
-    IO::out8(0x3d4, 0x0e);
-    IO::out8(0x3d5, MSB(value));
-    IO::out8(0x3d4, 0x0f);
-    IO::out8(0x3d5, LSB(value));
+    m_vga_arbiter->set_vga_text_mode_cursor({}, console_width, x, y);
+}
+#else
+void GraphicsManagement::set_vga_text_mode_cursor()
+{
 }
+#endif
```

Comment:
There is no corresponding `#if` in the `GraphicsManagement.h`. Could also move the `#if ARCH(..` into the function instead.